src/v/security/authorizer.h

```diff
@@ -26,6 +26,119 @@
 
 namespace security {
 
+/**
+ * Holds authZ check metadata for audit processing
+ */
+struct auth_result {
+    // Flag indicating if user is authorized
+    bool authorized{false};
+    // Indicates if the authorization system is disabled
+    bool authorization_disabled{false};
+    // Indicates if the user is a superuser
+    bool is_superuser{false};
+    // Indicates if no ACL matches were found
+    bool empty_matches{false};
+
+    // If found, the resource pattern that was matched to provide authZ decision
+    std::optional<std::reference_wrapper<const resource_pattern>>
+      resource_pattern;
+    // If found, the ACL that was matched that provided the authZ decision
+    std::optional<acl_entry_set::const_reference> acl;
+    // The principal that was checked
+    security::acl_principal principal;
+    // The host
+    security::acl_host host;
+    // The type of resource
+    security::resource_type resource_type;
+    // The name of the resource
+    ss::sstring resource_name;
```

Comment:
doesn't the caller already have this information?

Reply:
I think you're right on the two `resource_*` members.  The `principal` and `host` and 'discovered' further down the call stack from `request_config::authorized`